mozilla_django_oidc_db/middleware.py

```diff
@@ -1,14 +1,13 @@
-from mozilla_django_oidc.middleware import SessionRefresh
+from mozilla_django_oidc.middleware import SessionRefresh as _SessionRefresh
 from mozilla_django_oidc.utils import import_from_settings
 
 from .mixins import SoloConfigMixin
 from .models import OpenIDConnectConfig
 
 
-class SessionRefresh(SoloConfigMixin, SessionRefresh):
+class SessionRefresh(SoloConfigMixin, _SessionRefresh):
     def process_request(self, request):
-        config = OpenIDConnectConfig.get_solo()
-        if not config.enabled:
+        if not self.config.enabled:
```

Comment:
middleware classes are instantiated only once at the django startup phase, and the same instance is re-used for multiple requests. This means that at some point your `self.config` will be cached locally in python, and changes made in the admin will not be reflected.

I don't think you can use `self.config` here, but instead should keep the `get_solo` call.

This does introduce some challenges if you want to keep using the `SoloConfigMixin`, so maybe a `self.refresh_config()` method would be applicable, which internally does something like:

```py
def refresh_config(self):
    if hasattr(self, "_solo_config"):
        del self._solo_config
    self.config  # access to re-populate the cache (opitonal, as just accessing `self.config` in other code will have the same effect)
```

Reply:
Wouldn't it work if I implemented `refresh_config` and then do this in the middleware?

```python
class SessionRefresh(SoloConfigMixin, _SessionRefresh):
    def process_request(self, request):
        self.refresh_config()
        if not self.config.enabled:
            return

        return super().process_request(request)
```

Reply:
yep, it would - that's why I suggested it. Sorry if my rambling was not clear :D